nexus/src/app/instance.rs

```diff
@@ -1157,17 +1166,48 @@ impl super::Nexus {
             .fetch()
             .await?;
 
-        self.instance_ensure_dpd_config(
-            opctx,
-            db_instance.id(),
-            &sled.address(),
-            None,
-        )
-        .await?;
+        let boundary_switches =
+            self.boundary_switches(&self.opctx_alloc).await?;
+
+        for switch in &boundary_switches {
+            let dpd_client = self.dpd_clients.get(switch).ok_or_else(|| {
+                Error::internal_error(&format!(
+                    "could not find dpd client for {switch}"
+                ))
+            })?;
+            self.instance_ensure_dpd_config(
+                opctx,
+                db_instance.id(),
+                &sled.address(),
+                None,
+                dpd_client,
+            )
+            .await?;
+        }
 
         Ok(())
     }
 
+    // Switches with uplinks configured and boundary services enabled
+    async fn boundary_switches(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<HashSet<SwitchLocation>, Error> {
+        let mut boundary_switches: HashSet<SwitchLocation> = HashSet::new();
+        let uplinks = self.list_switch_ports_with_uplinks(opctx).await?;
+        for uplink in &uplinks {
+            let location: SwitchLocation =
+                uplink.switch_location.parse().map_err(|_| {
```

Comment:
Unfortunately this will require a bit of work to make `diesel` happy, so I created a follow up issue.
https://github.com/oxidecomputer/omicron/issues/3594